apps/server/src/modules/tool/launch-tool/service/strategy/oauth2-tool-launch.strategy.spec.ts

```diff
@@ -0,0 +1,65 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { ContextExternalToolDO, ExternalToolDO, SchoolExternalToolDO } from '@shared/domain';
+import { contextExternalToolDOFactory, externalToolDOFactory, schoolExternalToolDOFactory } from '@shared/testing';
+import { LaunchRequestMethod, PropertyData } from '../../types';
+import { OAuth2ToolLaunchStrategy } from './oauth2-tool-launch.strategy';
+import { IToolLaunchParams } from './tool-launch-params.interface';
+
+describe('OAuth2ToolLaunchStrategy', () => {
+	let module: TestingModule;
+	let strategy: OAuth2ToolLaunchStrategy;
+
+	beforeEach(async () => {
+		module = await Test.createTestingModule({
+			providers: [OAuth2ToolLaunchStrategy],
+		}).compile();
+
+		strategy = module.get(OAuth2ToolLaunchStrategy);
+	});
+
+	describe('buildToolLaunchRequestPayload', () => {
+		describe('when always', () => {
```

Comment:
when always in this line and line 30, 57 sound that you want to add more later and still missing.

Reply:
I just tried to follow the test structure. Do we make exceptions for things that don't have a condition?

Reply:
For me is only the description unclear ...what is mean with "when always" this 3 times. Is it is possible to add a more speaking when description?